tilelang/language/builtin.py

```
@@ -947,6 +947,35 @@ def sync_threads(barrier_id: int = None, arrive_count: int = None):
+    This is useful in warp-specialized producer/consumer pipelines:
+
+    .. code-block:: python
+
+        # Producer warp group: signal readiness, keep going
+        T.named_barrier_arrive(ready_barrier, total_threads)
+
+        # Consumer warp group: block until producer has arrived
+        T.sync_threads(ready_barrier, total_threads)
+
+    Args:
+        barrier_id:   Named barrier index (0–15). May be a variable (PrimExpr).
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)